sdks/java/core/src/main/java/org/apache/beam/sdk/testing/BigqueryMatcher.java

```diff
@@ -151,9 +171,34 @@ public void describeTo(Description description) {
 
   @Override
   public void describeMismatchSafely(PipelineResult pResult, Description description) {
-    description
-        .appendText("was (")
-        .appendText(actualChecksum)
-        .appendText(")");
+    String info;
+    if (response == null) {
```

Comment:
To provide more useful information here, try combining the first 3 cases into one via chained ||, and provide info of the form "Invalid BigQuery Response: %s.", Objects.toString(response).